src/session/query.c

```diff
@@ -198,9 +210,12 @@ int8_t _z_trigger_query_reply_partial(_z_session_t *zn, const _z_reply_context_t
                     (void)memset(&partial_reply, 0,
                                  sizeof(_z_reply_t));  // Avoid warnings on uninitialised values on the reply
                     partial_reply.data.sample.keyexpr = _z_keyexpr_duplicate(&reply.data.sample.keyexpr);
-                    pen_rep->_reply = partial_reply;
+                    pen_rep->_reply = _z_reply_new(partial_reply);
                 } else {
-                    pen_rep->_reply = reply;  // Store the whole reply in the latest mode
+                    pen_rep->_reply = _z_reply_new(reply); // Store the whole reply in the latest mode
+                }
+                if (pen_rep->_reply == NULL) {
+                    ret = _Z_ERR_SYSTEM_OUT_OF_MEMORY;
                 }
```

Comment:
if it fails due to out of memory, we should clean any function state and go to the return point.
Here the timestamp is duplicating as well, and it will still be a problem with out of memory.